### DISCUSSION TITLE
Koriki/Batocera Lite for RG35XX 1.0.3 - 20240103

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/Koriki#95] [Announcements]
This is version 1.0.3 (20240103) of Koriki/Batocera Lite for the RG35XX. This version includes among other things the possibility of changing launchers and the main capability of batocera, USB for gamepads and wireless USB dongles, and consistent charging and leds behavior for all hardware revisions. See the changes below to know more.

### Some changes:

2024/01/03:
* Fixed Shutdown from EmulationStation
* Fixed volume level changing upon entering/exiting emulators
* Fixed volume level being saved during reboots
* Fixed brightness level saved during reboots
* Fixed standalone emulators not launching correctly from EmulationStation
* Fixed PPSSPP and Drastic (NDS) not starting
* Added application shortcuts to enable/disable ADB, also to switch to EmulationStation or Simplemenu
* Added back batocera EmulationStation public domain content and associated art
* Additional internal boot process changes, etc.
* Fixed SDCard expansion not working on 20231220 release
* Added additional tools thumbnails (thanks https://github.com/SzalikDesigns)
* Fixed Error 256
* Added customlauncher file, so no need to add it manually (also launcher can now be changed via scripts)

2023/12/20:
* Fixed reboot/shutdown from menu (simplemenu)
* Added ``system/logs`` folder during installation. If you are reusing an existing card and have issues with error 256, see the important note above and create the folder manually.

2023/12/17:
* Included both simplemenu and EmulationStation launchers
* Changed completely the kernel and init mechanism to resemble the stock one:
  * Leds should now work consistently
  * Battery should report charge correctly
  * Battery charge icon is now displayed when the unit is connected to the power
  * This version should work with all RG35XX hardware revisions
* USB port should now work:
  * USB gamepads should work, although those can only be mapped in Simplemenu
  * USB memories also work
  * USB Wifi adapters now work. Currently only EDIMAX EW-711Un (rtl8192cu based) and TP-Link W725N (rtl8188us/eu based) work. You will need to use EmulationStation to configure the wireless network
* ADB can be disabled/enabled via Simplemenu menu

### Installation:

New installation:
* Download the batocera-rg35xx-rg35xx--20240103.img.gz file
* Using balena etcher or a similar tool flash the image to a SDcard (no need to extract the gz archive)
* Once the SDCARD has been flashed, insert the card into the first TF slot of the RG35XX and boot
* The first boot will take a bit longer since the system is expanding the SDCARD and also copying the default content to the last partition.

Update from previous release (1.0 to 1.0.2):
* You need to have koriki 1.0 or 1.0.1 installed on a SDCARD
* Mount the SDCARD on your computer
* Extract the (boot.tar.xz) archive from this release 
* From your original SDCARD, delete the file boot/batocera and copy the contents of the extracted boot.tar.xz into the first partition of the SDCARD (you should see Batocera-conf.conf, boot, etc.)

### Important:
* If you are updating from a previous version or you are using an existing SDCard on the TF2 slot, please remove or move your existing system folder (from the SHARE partition of the second card or the first card if you only use one).

Report findings, comments, and/or suggestions in the discussion page for this release

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_batocera_rg35xx_20240102'>Koriki/Batocera Lite for RG35XX 1.0.3 - 20240103</a>.</em>

**Comment by codevski** [comment]
I might be blind trying to find any information on it throughout the wiki or readme docs but cant find a answer. Just wondering does this support the rg35xx+ ? or these builds purely for rg35xx?

**Reply by acmeplus** [reply]
No, these builds are for the original RG35XX. The last preliminary build for the plus is [here](https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_plus_batocera_koriki_lite_pre-alpha_20231220).

I should get back to the plus soon.

**Reply by hoobminator** [reply]
can any one help i have done the new update for Koriki/Batocera Lite for RG35XX 1.0.3 - 20240103 but it will not launch it just stays on the same screen thanks dan

**Comment by Archix221** [comment]
How to set wifi dongle? I choose wifi settings but all freez, help reset.
All tests ports games like Shovel Knight, Curse of Issyos work fine on 1.0.3, but  AM2R dont start.

**Reply by acmeplus** [reply]
Wifi settings don’t work in koriki. You need to switch to emulationstation to configure wifi. Once you do that you can switch back to koriki.

**Reply by Archix221** [reply]
Ok,looks like mine dongle does not work. Although it is listed as supported. Thank you.

**Reply by acmeplus** [reply]
Will check. I’ve heard some reports of some issues from some users.

**Reply by demichel93** [reply]
@acmeplus I can confirm, I own WN725N and it does not work, when it's plugged in to the console before powering it up the console takes a sweet long time to do anything, there are no lights on the wifi dongle itself. When turning on Wifi via Emulation Station and going back to simple menu/Koriki it says wifi0 is offline.

**Reply by acmeplus** [reply]
Yeah, I've just verified, the last build includes the wrong wireless kernel driver modules. That was fixed on a previous version but it looks like the change was lost. I'll fix that tonight.

**Reply by demichel93** [reply]
when can this fixed release be expected?

**Reply by REGoman** [reply]
When is the fix firmware coming out? I still can't catch wifi when I connect TL-WN725N.

**Comment by benni092** [comment]
Hi there! This release works realy good so far i have tested it. Good job man!
I have just one issue. I cant launch drastic in es.
Here is the log 
[es_launch_stderr.log](https://github.com/rg35xx-cfw/Koriki/files/13828610/es_launch_stderr.log)
It seems that es want to start the desmume core, but i cant change the emulator that es is using for nds. The menuentry is not here. I copied drastic.sh to the ports folder and it start without problems using this way.

**Reply by acmeplus** [reply]
I’ll look into that but you can create a es_systems_nds.cfg to override that and always launch drastic. I’ll post details later.

**Comment by Charlychaps** [comment]
Hello , 
Happy New Year all!
As SHARE is now in EXT4 format, do we need to copy roms trought usb cable?
I'm under windows and of course I can't get any access to SHARE
Many thanks

**Reply by acmeplus** [reply]
Yeah. I need to fix that error. But in the meantime you should be able to erase the partition and format in fat32.

**Reply by ElCirujaCOl** [reply]
I can't put roms in the SD because windows won't allow me to see the partition. Even with Disk Genius I can't assign it a letter.

**Reply by italianbeef12** [reply]
What I did - burn the image onto the SD card. Eject and reinsert the card. Fire up disk genius or any partition manager out there. Reformat the share partition to fat32 (so it's no longer EXT4). Put it into your 53xx and boot it up (so it configures the partition correctly). Now you're good to go. You can put it back into your PC and copy files over.

**Comment by acmeplus** [comment]
Minor update addressing Drastic not working from ES and partition issues with ext4 formatting has been posted:

https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_batocera_rg35xx_20240105

**Comment by Res777** [comment]
A great upgrade, another great step forward, thank you!
I have some questions that I was already wondering from the previous version:
Emulation station
-system settings>language(region)
Does it refer to the theme or to batocera?  and does this work?
Are the languages included or do I have to download them separately? because by selecting any language also with a multilanguage theme, nothing changes after restart (which isn't reboot but shutdown... I think it's a bug), system and menu are still in English.
And a last question:
Is there a way to go back to simplemenu from within emulationstation (without putting the SD card on the PC)?

**Reply by acmeplus** [reply]
Frankly I've never tested language change in ES. It should work automatically but I may have not compiled the translation files in. I'll check because I know several users mentioned that they could change the language but the setting was not saved.  

To switch from ES to simplemenu, go to ports and select Simplemenu. To switch from Simplemenu to ES, go to apps or ports, and select EmulationStation. No need to edit anything or take the sdcard out. If for any reason it does not work, the file is in /userdata/system/customlauncher, and it can be reviewed/edited with od-commander directly.

**Reply by Res777** [reply]
Thanks for the support. To go back to simple menu, I tried but I don't have simplemenu or commander on ports.
Do I have to make .sh files? what strings should I write inside?

**Reply by acmeplus** [reply]
Do you have your own second SDCARD or have modify the contents? By default the image will populate those files, including roms/ports/Simplemenu.sh but if there's already a folder there the system will skip that copy.

If you don't have it, or you don't want to remove your existing system folder, you can create a new file called Simplemenu.sh in the roms/ports folder with this content:

```bash
#!/bin/sh

echo "simplemenu" > /userdata/system/customlauncher

killall -9 retroarch-launcher.sh simplemenu retroarch emulationstation

retroarch-launcher.sh
```

**Reply by Res777** [reply]
Thanks, so every time I upgrade/new install, the external sd must be empty? I understood that only system folder needed a backup for correct new populating

**Reply by acmeplus** [reply]
Normally you just need to move/rename your system folder since some config files may need to be updated. The rest of the sdcard with roms, bios, etc is not impacted. You can compare the files from the old system folder and new one after the first time it completely boots the system.

**Comment by benni092** [comment]
Unfortunaly drastic doesnt launch with the new update.
Here is the log:
[es_launch_stderr.log](https://github.com/rg35xx-cfw/Koriki/files/13849485/es_launch_stderr.log)
It seems that something wrong with the language function....
edit: i have noticed that the file /system/configs/drastic/config/drastic.cfg has 0bytes. something has it overwritten?
edit2: Jeah i coied the config file from an older card and drastic starts normal in simplemenu. as soon as i start drastic via es the config gets overwritten with an empty file.

**Reply by acmeplus** [reply]
It may have been an incomplete copy or something like that. You can just delete the configs/drastic and it should be recreated the next time it boots. Will check why the version copied has zero size. Most likely the original file where it’s been copied from is incorrect.

**Comment by luci-11** [comment]
hi,I can confirm wifi issue, also for the edimax dongle. 
Anyway, is it possible to enable netplay?  I tried putting global.netplay=1 in batocera-boot.conf without luck.
Thanks a lot

**Comment by ZuroFe** [comment]
HDMI only show boot logo.

Hello, thank you for your amazing work. ES is very amazing with many customization and nice music.   
There is an issue when using HDMI both in Koriki and ES. It only show the boot logo. This also happens in koriki 1.0.2.

**Reply by benni092** [reply]
Can confirm that....
![IMG_20240114_095420](https://github.com/rg35xx-cfw/Koriki/assets/122157977/e4163954-8bf6-4188-a690-e52a599be1af)

**Reply by AAcrazyman** [reply]
I can also confirm this. Same issue. Any fix yet?

**Reply by acmeplus** [reply]
Not yet

**Comment by remiz71** [comment]
Hello, I have wifi dongle TP-Link TL-WN823N (RTL8192CU based)
Could it’s working on 1.0.3?

**Reply by acmeplus** [reply]
Should work. Next update will fix wireless again.

**Comment by souporman** [comment]
1.0.3.1 - 20240105 says it doesn't format SHARE in EXT4 by default... but it still does?

EDIT: all versions as far back at v1 still format SHARE in EXT4.  If anyone needs to work around this, after I applied the image, and BEFORE I unplugged the microSD card, I used EaseUS Partition Master to change the 512MB SHARE partition from EXT4 to FAT32 and I left the unallocated space as unallocated.  Initial boot kept the SHARE partition as FAT32, and extended it to encompass the unallocated space.  I did this with v1, I didn't try it with the later builds, but I did update to those without issue.

**Reply by italianbeef12** [reply]
Thanks for sharing. I noticed the same thing but thought I was going crazy. :)

**Comment by Gui-RDS** [comment]
Help me, I flashed the image on the sd card, but when I try to boot, there is only a black screen. I tried reinstalling, pressing reset  and holding the turn on buttom and it doesn't work. How can I fix it? I even tried other OS and still get the same problem...
